cuegui/cuegui/Utils.py

```diff
@@ -257,8 +260,10 @@ def findJob(job):
     if not isinstance(job, six.string_types):
         return None
     if isStringId(job):
+        if re.search("^Job.", job):
+            job = re.sub("Job.", "", job)
         return opencue.api.getJob(job)
-    if not re.search(r"^([a-z0-9\_]+)\-([a-z0-9\.\_]+)\-", job, re.IGNORECASE):
+    if not re.search(r"^(?:Job.)?([a-z0-9\_]+)\-([a-z0-9\.\_]+)\-", job, re.IGNORECASE):
```

Comment:
You correctly used the `r` prefix to indicate that the string is a raw string. This is good practice, especially when your regex contains characters like `.`, `\` or `-` or `_`, as it prevents Python from interpreting them as escape sequences.

Suggestion: The regex is correctly used, but you can simplify it slightly:

`re.search(r"^(?:Job\.)?([a-z0-9_]+)-([a-z0-9._]+)-", job, re.IGNORECASE)`